src/class_resolver/docs.py

```diff
@@ -52,10 +86,32 @@ def f(
         from class_resolver.contrib.torch import activation_resolver, aggregation_resolver
 
         @document_resolver(
-            "activation", resolver_name="class_resolver.contrib.torch.activation_resolver"
+            DocKey("activation", "class_resolver.contrib.torch.activation_resolver"),
+            DocKey("aggregation", "class_resolver.contrib.torch.aggregation_resolver"),
         )
+        def f(
+            tensor: Tensor,
+            activation: Union[None, str, type[nn.Module], nn.Module],
+            activation_kwargs: dict[str, Any] | None,
+            aggregation: Union[None, str, type[nn.Module], nn.Module],
+            aggregation_kwargs: dict[str, Any] | None,
+        ):
+            _activation = activation_resolver.make(activation, activation_kwargs)
+            _aggregation = aggregation_resolver.make(aggregation, aggregation_kwargs)
+            return _aggregation(_activation(tensor))
+
+    If you have the resolvers handy, you can directly pass them.
+
+    .. code-block:: python
+
+        from typing import Any, Union
+        from torch import Tensor, nn
+        from class_resolver import document_resolver
+        from class_resolver.contrib.torch import activation_resolver, aggregation_resolver
+
         @document_resolver(
-            "aggregation", resolver_name="class_resolver.contrib.torch.aggregation_resolver"
+            DocKey("activation", activation_resolver),
```

Comment:
How do you get the fully qualified import name for the reference in the doc here?

Reply:
nvm, I found the NotImplemented part 😉

Reply:
that's a good question. I'm thinking we need to do some frame hacking 

@scolby33 get in on this

Reply:
gonna follow up with this later